Update some devDependencies, `npm audit fix`

tomalec opened the pull request:
### Changes proposed in this Pull Request:

Fixes some vulnerabilities reported by npm. 

```
fixed 22 of 39 vulnerabilities in 3064 scanned packages
  14 vulnerabilities required manual review and could not be updated
  1 package update for 3 vulnerabilities involved breaking changes
```

Bumps two devDependencies, to reduce our technical debt = dependency lag.

### Screenshots:

#### After
![image](https://user-images.githubusercontent.com/17435/147939526-3f2e040e-baae-4555-b19c-0dab5f19ce00.png)



### Detailed test instructions:

1. 
   ```
   npm install
   npm run test:js
   npm run lint:js
   ```
2. Only test-related dependencies were bumped, but you may want to smoke test a little to check if `npm audit fix` didn't break something. 


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

tomalec committed:
Update some devDependencies:

- `@testing-library/jest-dom@5.16.1` (from `5.15.0`)
- `@types/jest@27.4.0` (from `26.0.23`)
- `eslint-plugin-import@2.25.4` (from `2.23.4`)

tomalec committed:
Run `npm audit fix`.

tomalec committed:
Merge branch 'develop' into update/npm

eason9487 reviewed (approved):
LGTM.

eason9487 commented:
It doesn't relate to this PR but the **.externalized.json** file will be changed to add `@woocommerce/customer-effort-score` after running `npm start`. Could that change be committed along with this PR by the way?

tomalec committed:
Update `.externalized.json`,

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1178#issuecomment-1005580910
